apiserver/facades/client/application/deployrepository.go

```diff
@@ -677,7 +671,9 @@ func (v *deployFromRepositoryValidator) deducePlatform(ctx context.Context, arg
 	// Check that the placement platform and the derived platform match
 	// when a base is supplied. There is no guarantee that all placement
 	// directives are machine scoped.
-	if placementPlatform.String() == platform.String() {
+	if (platform.OS == "" || platform.OS == placementPlatform.OS) &&
+		(platform.Channel == "" || platform.Channel == placementPlatform.Channel) &&
+		(platform.Architecture == "" || platform.Architecture == placementPlatform.Architecture) {
 		return *placementPlatform, usedModelDefaultBase, nil
 	}
```

Comment:
`platformPlacement` doesn't have to be exactly equal to be valid, it just needs to be equal for the values set in `platform`